src/Experimental/RAGTools/preparation.jl

```diff
@@ -719,31 +720,31 @@ function build_index(
     return index
 end
 
-using Pinecone: Pinecone, init_v3, Index
+using Pinecone: Pinecone, PineconeContextv3, PineconeIndexv3, init_v3, Index
+# TODO: change docs
 """
     build_index(
-        indexer::PTPineconeIndexer;
+        indexer::PineconeIndexer;
         namespace::AbstractString,
-        schema::AbstractPromptSchema = OpenAISchema();
         verbose::Integer = 1,
         index_id = gensym("PTPineconeIndex"),
         cost_tracker = Threads.Atomic{Float64}(0.0))
 
-Builds a `PTPineconeIndex` containing a Pinecone context (API key, index and namespace).
+Builds a `PineconeIndex` containing a Pinecone context (API key, index and namespace).
 """
 function build_index(
-        indexer::PTPineconeIndexer,
+        indexer::PineconeIndexer,
+        context::Pinecone.PineconeContextv3 = Pinecone.init_v3(""),
+        index::Pinecone.PineconeIndexv3 = "",
         namespace::AbstractString,
-        schema::PromptingTools.AbstractPromptSchema = PromptingTools.OpenAISchema();
         verbose::Integer = 1,
-        index_id = gensym("PTPineconeIndex"),
+        index_id = gensym(namespace),
         cost_tracker = Threads.Atomic{Float64}(0.0))
+    @assert !isempty(context.api_key) && !isempty(index) "Pinecone context and index not set"
 
```

Comment:
As discussed, could we please add the chunking and embedding in `build_index`?

If people want to use an existing Pinecone index for querying, they should just use
`index = PineconeIndex(; id = index_id, context, index, namespace)` directly 
(we should mention it in the docs)

Reply:
Yes, definitely. I left a TODO for that and I'll add it soon.